src/app/data-planes/sources.ts

```diff
@@ -50,12 +53,20 @@ export const sources = (api: KumaApi) => {
       })
     },
 
-    '/meshes/:mesh/dataplane-overviews/:name': (params: DetailParams, source: Closeable) => {
+    '/meshes/:mesh/dataplanes/:name': async (params: DetailParams, source: Closeable) => {
       source.close()
 
       const { mesh, name } = params
 
-      return api.getDataplaneOverviewFromMesh({ mesh, name })
+      const [dataplane, dataplaneOverview] = await Promise.all([
+        api.getDataplaneFromMesh({ mesh, name }),
+        api.getDataplaneOverviewFromMesh({ mesh, name }),
+      ])
+
+      return {
+        dataplane,
+        dataplaneOverview,
+      }
```

Comment:
I’ve changed that now.

Reply:
Just wanted to note that preferably we should always use separate DataSource components instead of using `Promise.all` in the sources.ts file. There will probably the odd occasion where it make sense to use `Promise.all` in the `sources.ts` file and even then we should think carefully about whether it makes sense to or not.